src/theory/uf/ho_extension.cpp

```diff
@@ -42,6 +42,12 @@ HoExtension::HoExtension(Env& env,
       d_uf_std_skolem(userContext())
 {
   d_true = NodeManager::currentNM()->mkConst(true);
+  if (d_env.getLogicInfo().isHigherOrder()
```

Comment:
This is not the right place for this, options should never be modified after the SMT engine is initialized.

We should either move this code to `smt/set_defaults.cpp` or simply set the default value of the option to `true` in `options/uf_options`.

Reply:
Modified options/uf_options